src/console/Console.cxx

```diff
@@ -299,8 +302,8 @@ StateFlowBase::Action Console::Session::process_read()
             /* parse the line input into individual arguments */
             unsigned argc = 0;
             char last = '\0';
-
-            for (size_t i = 0; i < pos; ++i)
+            
+            for (size_t i = start; i < pos; ++i)
```

Comment:
I don't think this change is correct code.
When we find an end of line, we have to parse the entire line, just just the bytes that came since the last select call.
I don't know what is the issue you are trying to fix, but this does not seem like a good fix for it. A unittest that produces the problem first would be helpful.